editor/src/components/editor/actions/actions.ts

```diff
@@ -1944,41 +1945,49 @@ export const UPDATE_FNS = {
     )
   },
   INSERT_JSX_ELEMENT: (action: InsertJSXElement, editor: EditorModel): EditorModel => {
-    const editorWithAddedImport = modifyOpenParseSuccess((success) => {
-      const updatedImports = mergeImports(success.imports, action.importsToAdd)
-      return {
-        ...success,
-        imports: updatedImports,
-      }
-    }, editor)
-
-    const newSelectedViews: TemplatePath[] = []
-
-    const withNewElement = modifyOpenJSXElements((elements) => {
-      const targetParent =
-        action.parent == null
-          ? // action.parent == null means Canvas, which means storyboard root element
-            getStoryboardTemplatePath(elements)
-          : action.parent
+    let newSelectedViews: TemplatePath[] = []
+    const withNewElement = modifyUnderlyingTarget(
+      action.parent,
+      forceNotNull('Should originate from a designer', editor.canvas.openFile?.filename),
```

Comment:
is this inserting to the currently open file?

Reply:
No, it's starting the search from the currently open canvas file.